server/src/main/java/org/apache/druid/server/coordinator/loading/HttpLoadQueuePeon.java

```diff
@@ -223,7 +196,12 @@ private void doSegmentManagement()
     }
 
     final SegmentLoadingMode loadingMode = loadingModeSupplier.get();
-    int batchSize = calculateBatchSize(loadingMode);
+    final int batchSize = calculateBatchSize(loadingMode);
+
+    if (batchSize < 1) {
+      log.error("Batch size must be greater than 0.");
+      throw new RE("Batch size must be greater than 0.");
+    }
```

Comment:
We should retain the validation in this class but let's not throw an exception here.
The validation can move to `calculateBatchSize` method so that we always return a positive integer from there.

Also, please ensure that there are required validations in `SegmentLoadingConfig` and `HttpLoadQueuePeonConfig` around `numLoadingThreads` , `numBootstrapThreads` and `batchSize` respectively.

Reply:
Is the idea to have validated configs so that we validate it on startup instead of when we need the config during runtime?

Looking at these config classes, that doesn't seem to be the convention, there is no validation for any other config, and some config classes `SegmentLoaderConfig` would require some refactor to have a constructor to do such validation on init.

Reply:
Yes, configs should be validated on startup itself. There is not much use doing the validation and throwing an exception when we use the config values.

You can create a follow up PR to do the validation.